### DISCUSSION TITLE
How to convert the Spectral distribution of an LED to an illuminant?

### DISCUSSION THREAD

**Discussion by volkerjaenisch** [colour-science/colour#783] [Q&A] [answered]
Hi People of colour-science!
Thanks for these really cool libraries.

I am quite new to colour, so please excuse my stupid question.
I have a spectroscope measurement of the spectral intensity of a "white" LED. How can I build a Illumninant Instance from this data to use it in python-colour?

Cheers,
Volker

**Comment by thomasmansencal** [comment, marked as answer]
Hi @volkerjaenisch,

No such a thing as a stupid question! Assuming your data is tabulated, you would simply need to instantiate it with a `colour.SpectralPowerDistribution` class, for example:

```python
import colour

data = {
    500: 0.0651,
    520: 0.0705,
    540: 0.0772,
    560: 0.0870,
    580: 0.1128,
    600: 0.1360
}
sd_illuminant = SpectralDistribution(data, interpolator=colour.LinearInterpolator)
```

The tutorial has some more info about this class: https://colour.readthedocs.io/en/develop/tutorial.html#from-spectral-distribution

When you have your illuminant spectral distribution, you can then use it with `colour.sd_to_XYZ`, e.g. `colour.sd_to_XYZ(my_sd_reflectance, illuminant=sd_illuminant)`
